tests/test_triton_server_factory.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from .common import test_result_collector as trc
+
+from model_analyzer.triton.server.server_factory import TritonServerFactory
+from unittest.mock import patch, MagicMock
+from .mocks.mock_config import MockConfig
+from .mocks.mock_os import MockOSMethods
+
+from model_analyzer.config.input.config_command_profile import ConfigCommandProfile
+from model_analyzer.cli.cli import CLI
+
+
+class TestTritonServerFactory(trc.TestResultCollector):
+
+    def setUp(self):
+        # Mock path validation
+        self.mock_os = MockOSMethods(
+            mock_paths=['model_analyzer.triton.server.server_factory'])
+        self.mock_os.start()
+
+    def _test_get_server_handle_helper(self, launch_mode, expect_local,
+                                       expect_docker, expect_config, use_dcgm):
+        """
+        Test get_server_handle() calls the correct Triton Server create function with the
+        correct Triton Server Config options
+
+        Parameters:
+        ----------
+        launch_mode: str
+            which launch mode to use
+        use_dcgm: bool
+            If true, set use_local_gpu_monitor in the input config
+        expect_local: bool
+            If true, assert that create_server_local() is called
+        expect_docker: bool
+            If true, assert that create_server_docker() is called
+        expect_config: bool
+            If true, verify that certain Triton Server args are passed to the create function as expected
+            If false, verify that those args are NOT passed
+        """
+
+        config = ConfigCommandProfile()
+        config.triton_launch_mode = launch_mode
+        config.triton_http_endpoint = "1:2"
+        config.triton_grpc_endpoint = "3:4"
+        config.monitoring_interval = 0.5
+        config.use_local_gpu_monitor = use_dcgm
+
+        expected_http_port = '2'
+        expected_grpc_port = '4'
+        expected_metrics_interval_ms = 500
```

Comment:
Is there a reason why the expected metrics interval is 500 (maybe a default)? I feel like we need to document this in a comment.